nima/webclient/webclient/src/main/java/io/helidon/nima/webclient/http1/Http1ClientConnection.java

```diff
@@ -105,33 +110,63 @@ boolean isConnected() {
         return socket != null && socket.isConnected();
     }
 
+    private InetSocketAddress inetSocketAddress() {
+        DnsResolver dnsResolver = connectionKey.dnsResolver();
+        if (dnsResolver.useDefaultJavaResolver()) {
+            return new InetSocketAddress(connectionKey.host(), connectionKey.port());
+        } else {
+            InetAddress address = dnsResolver.resolveAddress(connectionKey.host(), connectionKey.dnsAddressLookup());
+            return new InetSocketAddress(address, connectionKey.port());
+        }
+    }
+
+    private int connectToProxy(InetSocketAddress remoteAddress) throws IOException {
+        String hostPort = remoteAddress.getHostName() + ":" + remoteAddress.getPort();
+        // FIXME Ideally we would pass the socket of this connection into the next Http1Client
+//        Http1Client proxyConnection = Http1Client.builder()
+//                .baseUri("http://" + hostPort).build();
+//        Http1ClientResponse response = proxyConnection.method(Method.CONNECT)
+//                .header(Header.HOST, hostPort).request();
+//        return response.status().code();
+
+        // FIXME This has to be replaced by previous lines
+        String newLine = "\r\n";
+        HelidonSocket tempSocket = PlainSocket.client(socket, createChannelId(socket));
+        StringBuilder httpConnect = new StringBuilder();
+        httpConnect.append("CONNECT ").append(hostPort).append(" HTTP/1.1").append(newLine)
+        .append("Host: ").append(hostPort).append(newLine)
+        .append("Accept: */*").append(newLine).append(newLine);
+        // TODO Authentication
+        if (LOGGER.isLoggable(DEBUG)) {
+            LOGGER.log(DEBUG, String.format("Proxy client connected %s %s",
+                                            connectionKey.proxy().address(),
+                                            Thread.currentThread().getName()));
+        }
+        DataWriter writer = SocketWriter.create(null, tempSocket, -1);
+        BufferData data = BufferData.create(httpConnect.toString().getBytes(StandardCharsets.UTF_8));
+        writer.writeNow(data);
+        DataReader reader = new DataReader(tempSocket);
+        int statusCode = readStatusCode(reader);
```

Comment:
Please use `Http1StatusParser` instead of this `readStatusCode` method.

Reply:
I was using that one, but that one does not support HTTP 1.0. I tested that there are some proxies that, even when you send the request for HTTP 1.1, they respond with HTTP 1.0.